spec/features/staff/users_can_create_a_programme_level_activity_spec.rb

```diff
@@ -103,6 +103,17 @@
 
         choose("activity[programme_status]", option: "07")
         click_button t("form.button.activity.submit")
+
+        if parent.roda_identifier_compound.include?("NF")
```

Comment:
I'm not sure I understand your question. I believe Leeky already added specifications on the[ factories](https://github.com/UKGovernmentBEIS/beis-report-official-development-assistance/pull/766/files#diff-3287c872813b2fc3d0cf92af0c553b243275ce0041c4163b580038b1f6ad4283) and [seeds](https://github.com/UKGovernmentBEIS/beis-report-official-development-assistance/pull/766/files#diff-f05850356b5d25098ac0b9272b7aab3e50fd47a67bdac0bf9419fb7caca2077f) so we can work with the correct data for each fund. Not sure if that's what you mean

Reply:
What I mean is that the only way we currently check that a fund is Newton or GCRF is by looking at the RODA identifier fragment. This feels like an implementation detail, something about the internals of an activity that could potentially change.

If that identifier changes, for example, or if in the future we'll have to identify it by title, we'll have to change every spec where we test that a programme's parent fund is Newton or GCRF.

Having a method on the activity that checks if the activity is a fund and it has the expected RODA identifier fragment, then using that method everywhere, would mean we only have one place where the detail about the fragment is contained.

It could also be overkill, I don't know, but it would be clearer about the meaning and intention of the check.

If I'm new to the project and read `parent.roda_identifier_compound.include?("NF")`, I won't know what it means, but if I read `parent.is_newton_fund?` I'll know the meaning of what I'm reading.

Reply:
See comment above re: not using `parent.is_newton_fund?`